apps/lunch_together/endpoints/v1/appointments.py

```diff
@@ -0,0 +1,201 @@
+"""
+CRUD of Lunch Together appointments.
+Path: /apps/lunch-together/v1/appointments/
+"""
+
+from datetime import datetime
+from typing import List
+
+from bson import ObjectId
+from bson.errors import InvalidId
+from fastapi import status, APIRouter, Depends
+from fastapi.responses import JSONResponse
+from pymongo.errors import PyMongoError
+
+from apps.user.models.auth import User
+from apps.user.models.message import Message
+from apps.user.services.auth_service import auth_user
+from services.mongo_service import get_mongo
+from ...models.appointments import Appointment
+
+router = APIRouter()
+
+
+@router.get(
+    '/',
+    dependencies=[Depends(auth_user)],
+    response_model=List[Appointment],
+    responses={
+        status.HTTP_500_INTERNAL_SERVER_ERROR: {
+            'model': Message,
+        },
+    },
+)
+def get_appointment(database=Depends(get_mongo)):
+    """
+    Get an appointment whose datetime is after now.
+    """
+    try:
+        appointments = list(
+            database.lunch_appointments.find({
+                'datetime': {
+                    '$gt': datetime.now(),
+                },
+            }))
+        return appointments
+    except PyMongoError as error:
+        return JSONResponse(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            content={'message': str(error)},
+        )
+
+
+@router.post(
+    '/',
+    responses={
+        status.HTTP_201_CREATED: {
+            'model': Message,
+        },
+        status.HTTP_500_INTERNAL_SERVER_ERROR: {
+            'model': Message,
+        },
+    },
+)
+def post_appointment(
+        appointment: Appointment,
+        database=Depends(get_mongo),
+        user: User = Depends(auth_user),
+):
+    """
+    Post an appointment.
+    """
+    try:
+        database.lunch_appointments.insert_one({
+            'restaurant_id': appointment.restaurant_id,
+            'title': appointment.title,
+            'datetime': appointment.datetime,
+            'n_participants': appointment.n_participants,
```

Comment:
Is there no need for the participants' names?

Reply:
I intentionally omitted it in order to see you actually code review :) Just kidding. I'll edit it right away.

Reply:
Having thought of it, how about making another collection called 'lunch_registrations' having participants' names and others, so that creating and canceling a registration be creating and deleting a document in that collection.

Reply:
sounds good